contrib/gofiber/fiber.v2/fiber_test.go

```diff
@@ -158,6 +158,34 @@ func TestCustomError(t *testing.T) {
 	assert.Equal(fiber.ErrBadRequest, span.Tag(ext.Error).(*fiber.Error))
 }
 
+func TestUserContext(t *testing.T) {
+	assert := assert.New(t)
+	mt := mocktracer.Start()
+	defer mt.Stop()
+
+	// setup
+	router := fiber.New()
+	router.Use(Middleware(WithServiceName("foobar")))
+
+	router.Get("/", func(c *fiber.Ctx) error {
+		// check if not default empty context
+		assert.NotEmpty(c.UserContext())
+		span, _ := tracer.StartSpanFromContext(c.UserContext(), "http.request")
+		defer span.Finish()
+		return c.SendString("test")
+	})
+	r := httptest.NewRequest("GET", "/", nil)
+
+	router.Test(r, 100)
+
+	// verify both middleware span and router span finished
+	spans := mt.FinishedSpans()
+	assert.Len(spans, 2)
+	if len(spans) < 2 {
+		t.Fatalf("no spans")
+	}
+}
```

Comment:
I think this will still pass even without the context change you made, won't it?

We probably want to assert that the spans have a parent/child relationship, which they will have iff the context is propagated correctly. See: https://pkg.go.dev/gopkg.in/DataDog/dd-trace-go.v1@v1.31.0/ddtrace/mocktracer#Span for comparing checking parentage.